model/eligibility/Eligibility.php

```diff
@@ -138,15 +149,35 @@ public function getId()
     }
 
     /**
-     * @return array
+     * @return Resource
      * @throws \core_kernel_persistence_Exception
      */
+    public function getProctorBypassed(){
+        if($this->proctored === null){
+            $this->proctored =$this->getProperty(EligibilityService::PROPERTY_BYPASSPROCTOR_URI);
```

Comment:
`$this->getProperty()` will always return `` here.
If you want to get property's value you need to get eligibility resource at first and after that get property value from this resource, something like that should work:
`$this->proctored = $this->getResource($this->id)->getOnePropertyValue($this->getProperty(EligibilityService::PROPERTY_BYPASSPROCTOR_URI));`

As a small performance improvement you can extract `$this->getResource($this->id)` into getEligibilityResource() method, or something like that, and cache this resource as a class property. Now we call `$this->getResource($this->id)` in 2 places and you will add one more call here.